gslib/tests/test_signurl.py

```diff
@@ -144,6 +158,7 @@ def testSignUrlWithJSONKeyFileAndObjectGeneration(self):
     self.assertIn('x-goog-credential=' + TEST_EMAIL, stdout)
     self.assertIn('generation=' + object_uri.generation, stdout)
 
+  @unittest.skipUnless(HAVE_CRYPTO, 'signurl requires crypto to decode .p12 keys.')
```

Comment:
This test doesn't look like it's covering anything specific to P12, so maybe change self._GetKsFile() to self._GetJSONKsFile()

Same with some of the tests below.

Reply:
That's right, but this test only covers URL encoding behavior -- that's not behavior we need a .p12 file to confirm. 

This concern is related to the one I raised above -- if for some reason the test environment shifts such that it doesn't have crypto, the behavior this test covers will never be verified since it will always be skipped. Removing the skip decorator here and switching to JSON keys means it's more likely to continue running.

To be clear, this comment did not apply to the test cases  above, like testSignUrlRaiseErrorForP12KeysWithoutCrypto or testSignUrlOutputP12, as those explicitly cover behavior related to p12 files.

Reply:
Done! Thanks for clarification